### PR TITLE
adds mime type validations for insured::inbox controller

### PR DESCRIPTION
**Pull request by TristanB17**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [x] Tests for the changes have been added (for bugfixes/features)

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/story/show/187465857

# A brief description of the changes

Current behavior: No mime-type validations for Insured::Inboxes controller

New behavior: Insured::Inboxes controller now will only respond to valid mime-types

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

### CODE REVIEW

**Review by ymhari — APPROVED** [review]

File: app/controllers/insured/inboxes_controller.rb [hunk 1]
```diff
@@ -18,6 +19,11 @@ def successful_save_path
     exchanges_hbx_profiles_root_path
   end
 
+  def destroy
```
Comment (ymhari) [inline app/controllers/insured/inboxes_controller.rb:22]: Why are we adding a new action?

**Review by ymhari — CHANGES_REQUESTED** [review]

File: app/controllers/insured/inboxes_controller.rb (hunk 1 above)
Comment (saikumar9) [inline app/controllers/insured/inboxes_controller.rb:22]: @TristanB17 Is there a reson for adding this endpoint?

**Review by saikumar9 — CHANGES_REQUESTED** [review]
- All the changes look good to me except for one. Please review my comment.

File: app/controllers/insured/inboxes_controller.rb (hunk 1 above)
Comment (TristanB17) [reply app/controllers/insured/inboxes_controller.rb]: @saikumar9 it was and endpoint being inherited from InboxesController, but the original endpoint didn't have any MIME type restrictions -> I removed the endpoint and added the MIME restrictions to the base controller after verifying all child controllers were only supposed to respond with JS

My mistake, I thought doing it the way I originally did it was easier to test/a more gradual change.

**Review by saikumar9 — APPROVED** [review]
lgtm!

**Review by ymhari — APPROVED** [review]

**Pull request merged**